src/components/views/CreateBounty.jsx

```diff
@@ -29,17 +29,80 @@ function CreateBounty(props) {
     state: { isForeignNetwork },
     actions: { displayForeignNetRequiredWarning },
   } = useContext(Web3Context);
-  const { id: campaignId, slug: campaignSlug } = props.match.params;
+  const { slug: campaignSlug, milestoneId } = props.match.params;
 
-  const campaign = useCampaign(campaignId, campaignSlug);
-  const [bounty, setBounty] = useState({
+  const [campaign, setCampaign] = useState();
+  const [donateToDac, setDonateToDac] = useState(true);
+  const [milestone, setMilestone] = useState({
     title: '',
     description: '',
-    recipientAddress: ZERO_ADDRESS,
-    picture: '/img/bountyProject.png',
     donateToDac: true,
     reviewerAddress: '',
   });
+  const [initialValues, setInitialValues] = useState({
+    title: '',
+    description: '',
+    donateToDac: true,
+    reviewerAddress: '',
+  });
+
+  const isNew = milestoneId === undefined;
+
+  function goBack() {
+    history.goBack();
+  }
+
+  const getCampaign = async slug => {
+    CampaignService.getBySlug(slug)
+      .then(res => {
+        setCampaign(res);
+      })
+      .catch(err => {
+        const message = `Sadly we were unable to load the requested Campaign details. Please try again.`;
+        ErrorHandler(err, message);
+      });
+  };
+
+  useEffect(() => {
+    if (isNew) {
+      getCampaign(campaignSlug).then();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (!isNew && currentUser.id) {
```

Comment:
What if user doesn't have any `id`? a user which has not signed in or created profile